torch/jit/frontend.py

```diff
@@ -346,17 +355,26 @@ def build_param_list(ctx, py_args, self_name):
             if arg is not None:
                 ctx_range = build_expr(ctx, arg).range()
                 raise NotSupportedError(ctx_range, _vararg_kwarg_err)
-    result = [build_param(ctx, arg, self_name, False) for arg in py_args.args]
-    result += [build_param(ctx, arg, self_name, True) for arg in py_args.kwonlyargs]
+
+    # List of Tuple of args and type as inferred by profile directed typing
+    arg_and_types = [(arg, next(iter(pdt_arg_types[arg.arg])) if bool(pdt_arg_types[arg.arg]) else None) for arg in py_args.args]
+    arg_and_types_kwonlyargs = [(arg, next(iter(pdt_arg_types[arg.arg])) if bool(pdt_arg_types[arg.arg]) else None)
```

Comment:
@gmagogsfm : Also `bool(pdt_arg_types[arg.arg])` - More pythonic way is to check the `None`ness. But a check on a empty `set` is treated as `True`, hence I am converting to a `bool`. This will go away when address the cleanup's in my other PR.